File: iOS/TestFibonacciObjectiveC/TestFibonacciObjectiveC/ViewController.m
//
//  ViewController.m
//  TestFibonacciObjectiveC
//
//  Created by AWL_02 on 03/07/2014.
//  Copyright (c) 2014 awl. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
            

@end

@implementation ViewController
            
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.

}

- (IBAction)buttonLaunchClick:(id)sender {
    NSDate *date1 = [[NSDate alloc] init];
    int number = [self fib:30];
    
    NSDate *date2 = [[NSDate alloc] init];
    
    NSTimeInterval diff = [date2 timeIntervalSinceDate:date1];
    NSString *intervalString = [NSString stringWithFormat:@"%f", diff];
    
    NSLog(@"%@", intervalString);
    NSLog(@"%d", number);
}


-(int) fib: (int)start {
    if (start == 0) {
        return start;
    }
    if (start == 1) {
        return start;
    }
    return ([self fib:start-1] + [self fib:start-2]);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: iOS/TestFibonacciSwift/TestFibonacciSwift/ViewController.swift
//
//  ViewController.swift
//  TestFibonacciSwift
//
//  Created by AWL_02 on 03/07/2014.
//  Copyright (c) 2014 awl. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
                            
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
    }
    
    @IBAction func buttonLaunchClick(sender: AnyObject) {
        let dateBeginning:NSDate = NSDate()
        var answer = fib(30)
        
        let dateEnd:NSDate = NSDate()
        let intervalSeconds:NSTimeInterval = dateEnd.timeIntervalSinceDate(dateBeginning)
        let result:NSString = NSString(format: "%f", intervalSeconds)
        
        println(answer)
        println(result)
    }
    
    func fib(start: Int) -> Int {
        if start == 0 {
            return start
        }
        if start == 1 {
            return start
        }
        return (fib(start - 1) + fib(start - 2))
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

